pkg/operator/k8sutil/k8sutil.go

```diff
@@ -70,3 +71,41 @@ func GetK8SVersion(clientset kubernetes.Interface) (*version.Version, error) {
 	}
 	return version.MustParseSemantic(serverVersion.GitVersion), nil
 }
+
+// MergeResourceRequirements merges two resource requirements together (first overrides second values)
+func MergeResourceRequirements(first, second v1.ResourceRequirements) v1.ResourceRequirements {
+	// if the first has a value not set check if second has and set it in first
+	if _, ok := first.Limits[v1.ResourceCPU]; !ok {
+		if _, e := second.Limits[v1.ResourceCPU]; e {
+			if first.Limits == nil {
```

Comment:
Perhaps we can remove checking if the element in the second struct exist. Something like this to make it simpler:

```
if _, ok := first.Limits[v1.ResourceCPU]; !ok {
    if first.Limits == nil {
        first.Limits = v1.ResourceList{}
    }
    first.Limits[v1.ResourceCPU] = second.Limits[v1.ResourceCPU]
}
```

So whether the second struct has value or not, it should be assigned to the first struct

Reply:
Yes that is better, changed it.